extensions/resteasy-reactive/quarkus-resteasy-reactive/runtime/src/main/java/io/quarkus/resteasy/reactive/server/runtime/ServerVertxAsyncFileMessageBodyWriter.java

```diff
@@ -0,0 +1,63 @@
+package io.quarkus.resteasy.reactive.server.runtime;
+
+import java.lang.reflect.Type;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.ext.Provider;
+
+import org.jboss.resteasy.reactive.server.core.ResteasyReactiveRequestContext;
+import org.jboss.resteasy.reactive.server.spi.ResteasyReactiveResourceInfo;
+import org.jboss.resteasy.reactive.server.spi.ServerHttpResponse;
+import org.jboss.resteasy.reactive.server.spi.ServerMessageBodyWriter;
+import org.jboss.resteasy.reactive.server.spi.ServerRequestContext;
+
+import io.quarkus.resteasy.reactive.common.runtime.VertxAsyncFileMessageBodyWriter;
+import io.vertx.core.Handler;
+import io.vertx.core.file.AsyncFile;
+
+@Provider
+public class ServerVertxAsyncFileMessageBodyWriter extends VertxAsyncFileMessageBodyWriter
+        implements ServerMessageBodyWriter<AsyncFile> {
+
+    @Override
+    public boolean isWriteable(Class<?> type, Type genericType, ResteasyReactiveResourceInfo target, MediaType mediaType) {
+        // allow for subtypes, such as AsyncFileImpl
+        return AsyncFile.class.isAssignableFrom(type);
+    }
+
+    @Override
+    public void writeResponse(AsyncFile file, Type genericType, ServerRequestContext context) throws WebApplicationException {
+        ResteasyReactiveRequestContext ctx = ((ResteasyReactiveRequestContext) context);
+        ctx.suspend();
+        ServerHttpResponse response = context.serverResponse();
+        response.setChunked(true);
```

Comment:
Why chunked? You should know exactly how big the file is so you can set the content length.

Reply:
Filed https://github.com/eclipse-vertx/vert.x/issues/3892